### DISCUSSION TITLE
Limit definitions

### DISCUSSION THREAD

**Discussion by domf4ju** [arianneorpilla/jidoujisho#178] [General]
Hi, not sure if I'm using the right terminology here but how can I limit the number of definitions/entries in the dictionary when I click on a word? I wanted to simplify it a bit.

**Comment by arianneorpilla** [comment]
You can't. Moreover, there should be no discernable search performance difference when hiding dictionaries. 

Is there a reason you'd like to do this? I recommend collapsing dictionaries by default instead. This will hide some dictionaries until you show them, which is useful if you want to hide bilingual/monolingual dictionaries.

**Reply by domf4ju** [reply]
It's not an issue with performance (it's very fast). It's that when clicking on a word I'm seeing 20+ entries. Even with them collapsed, it just takes up a lot of space. I'm aware I could totally be doing something wrong so let me know. Here is a quick screenshot - it's just the first part, but it would take a couple thumb flicks to reach the end of those collapsed sections. 
![Screenshot_20230420-181435](https://user-images.githubusercontent.com/60677130/233499095-31c61ca5-8d88-4592-b653-46ced2f4c549.png)

**Reply by arianneorpilla** [reply]
I would say it's pretty essential for you to be able to see every definition available to you, especially if you are at the point that you are looking up words like 出る. 

I don't think it's practical to hide definitions because it would essentially be limiting what you can find -- and for JMdict if you limit it to 5, then there's no telling if the actual context meaning is in the last, or in the 2/3 point, or wherever.

I think the easiest words will be the most extreme cases for this, where JMdict is gonna have 20+ entries for words like 赤 and する.

**Reply by arianneorpilla** [reply]
Limiting to a maximum number or fraction of entries for a single dictionary would be bad because you don't know which one is going to be the correct one, and limiting the maximum number to a certain dictionary is in my view, impractical because the user is able to control the number of dictionaries they want anyway, by way of collapsing, hiding or not importing too many in the first place.

I think this problem won't be as big of a deal when you're no longer looking up the simplest words. Let me know what you're thinking however.

**Reply by domf4ju** [reply]
Sorry that was just a random example I tapped on, but I think it answered my question! I didn't notice that it shows fewer entries for certain words. Yes I agree with you on those points. Thanks so much for the answer!
